src/components/Courses.js

```diff
@@ -1,8 +1,83 @@
-import React, {Component} from 'react'
-class Courses extends Component{
-    render(){
-        return <h1>Courses page</h1>
-    }
-    
+import React, { Component, useState } from 'react'
+import Assignment from './Assignment'
+import { Container, Card, Button,Row,Col, Text} from 'react-bootstrap';
+import CourseService from '../services/courseService';
+import 'bootstrap/dist/css/bootstrap.min.css';
+
+const Courses = () => {
+  const [courses, setCourses] = useState([]);
+
+  const [newCourse, setNewCourse] = useState({
+    title: "",
+    course_code: "",
+    session_code: "",
+    semester_code: "",
+  });
+  const newCourseChanged = (e) => {
+    setNewCourse({ ...newCourse, [e.target.name]: e.target.value });
+  }
+
+  return <Container><div><h1>Courses page</h1>
+  <Row><Col>
+ <button type="button" className="btn btn-primary" onClick={
+      async () => {
+        setCourses(await CourseService.getCourses());
+        console.log(courses[0]);
+      }
+    }>Display All Courses</button></Col>
+    </Row>
+    <br/>
+    <Row>
+    {courses.map((course) => {
+      return <Col><Card style={{width:'30rem' }}>
+        {/* <Card.Img variant="top" src="https://via.placeholder.com/30x30" /> */}
+        <Card.Body>
+          <Card.Title>Course Name</Card.Title>
+          <Card.Text><CourseDetails key={course.id} course={course} />
```

Comment:
what exactly does the `key={course.id}` portion of this do?  it doesn't appear that in the `CourseDetails` function below that the `key` is used, like the `course` property is

Reply:
To our services to work key={course.id} is not required just to make it work with future enhancement like  selecting the course details or selecting the exact assignments we may need it .